sdk/identity/identity/test/node/clientCertificateCredential.spec.ts

```diff
@@ -78,6 +79,53 @@ describe("ClientCertificateCredential", function() {
     }
   });
 
+  it("sends a correctly formatted token request with x5c enabled", async () => {
+    const tenantId = "tenantId";
+    const clientId = "clientId";
+    const mockHttpClient = new MockAuthHttpClient();
+    // Enable X5c flag
+    mockHttpClient.tokenCredentialOptions.includeX5c = true;
+
+    const credential = new ClientCertificateCredential(
+      tenantId,
+      clientId,
+      path.resolve(__dirname, "../test/azure-identity-test.crt"),
+      mockHttpClient.tokenCredentialOptions
+    );
+
+    await credential.getToken("scope");
+
+    const authRequest = mockHttpClient.requests[0];
+    if (!authRequest) {
+      assert.fail("No authentication request was intercepted");
+    } else {
+      assert.strictEqual(
+        authRequest.url.startsWith(`https://authority/${tenantId}`),
+        true,
+        "Request URL doesn't contain expected tenantId"
+      );
+      assert.strictEqual(
+        authRequest.body.indexOf(`client_id=${clientId}`) > -1,
+        true,
+        "Request URL doesn't contain expected clientId"
+      );
+
+
+      const queryParams = qs.parse(authRequest.body);
+      const jwtToken = jws.decode(queryParams.client_assertion as string);
+
+      assert.strictEqual(jwtToken.header.x5t, (credential as any).certificateX5t);
+      assert.deepStrictEqual(jwtToken.header.x5c, (credential as any).certificateX5c);
```

Comment:
Wouldn't this pass even when the credential's `certificateX5c` is invalid?

Reply:
What do you mean by invalid?

Reply:
Incorrect according to the RFC. This appears to verify only that the credential's `certificateX5c` value equals the JWT header's x5c value. If that's the case, then the test would pass even when the credential's `certificateX5c` is something AAD would reject as malformed.

Reply:
Added a chain test and check that it's the correct chain length.